transport/pool_integration_test.go

```diff
@@ -40,7 +42,36 @@ func TestNodeConnPoolIntegration(t *testing.T) {
 
 	t.Log("Close pool")
 	p.Close()
-	time.Sleep(250 * time.Millisecond)
+	time.Sleep(refillerBackoff)
+
+	t.Log("Check if connections are closed")
+	for _, c := range p.AllConns() {
+		if c != nil {
+			t.Fatalf("conn %s", c)
+		}
+	}
+}
+
+func TestCloseConnPoolIntegration(t *testing.T) {
```

Comment:
What is the purpose of this test i.e. what does it test?

Reply:
This test assures that no unwanted behavior happens if in between shard close requests there happens pool refiller close request.
This was a bug a while ago which this test helped to solve so I didn't remove it just in case, should it be kept?